routes/tp.js

```diff
@@ -339,4 +339,71 @@ router.put('/tp/:id/reserve/now', async function(req, res) {
     res.end();
 });
 
+router.put('/tp/:id/reserve/extend', async function(req, res) {
+    if(! req.locals.logInfo.is_logged) {
+        res.status(403).send({message: 'Not authorized'});
+        return;
+    }
+    if(! sansrv.sanitizeAll([req.params.id])) {
+        res.status(403).send({message: 'Invalid parameters'});
+        return;
+    }
+    let user = null;
+    let isadmin = false;
+    try {
+        user = await dbsrv.mongo_users().findOne({'_id': req.locals.logInfo.id});
+        isadmin = await rolsrv.is_admin(user);
+    } catch(e) {
+        logger.error(e);
+        res.status(404).send({message: 'User session not found'});
+        res.end();
+        return;
+    }
+
+    if(!user) {
+        res.status(404).send({message: 'User does not exist'});  // Why do none of the other routes give a 404 status here ?
```

Comment:
They probably should, but to be honest this is a real edge case: this mean someone is logged in, but somehow his account does not exists.

Not sure if this can actually happen